lib/matplotlib/cbook/__init__.py

```diff
@@ -2002,6 +2002,47 @@ def _array_perimeter(arr):
     ))
 
 
+def _unfold(arr, axis, size, step):
+    """Append an extra dimension containing sliding windows along ``axis``.
+
+    All windows are of size ``size`` and begin with every ``step`` elements.
+
+    Parameters
+    ----------
+    arr : ndarray, shape (N_1, ..., N_k)
+        The input array
+
+    Returns
+    -------
+    windows : ndarray, shape (N_1, ..., 1 + (N_axis-size)/step, ..., N_k, size)
+    """
+    new_shape = list(arr.shape) + [size]
+    new_strides = list(arr.strides) + [arr.strides[axis]]
+    new_shape[axis] = (new_shape[axis] - size) // step + 1
+    new_strides[axis] = new_strides[axis] * step
+    return np.lib.stride_tricks.as_strided(arr,
+                                           shape=new_shape,
+                                           strides=new_strides,
+                                           writeable=False)
+
+
+def _array_patch_perimeters(x, rstride, cstride):
+    """Extract perimeters of patches from ``arr``.
```

Comment:
argument name is wrong here (change either in the signature or the docstring).
also we use `*param_name*` as markup for parameters (not repeating the same comment throughout ;-))